src/util/FastMap.ts

```diff
@@ -1,23 +1,31 @@
 export class FastMap {
-  size: number = 0;
-  private _values: Object = {};
+  private _size: number = 0;
+  get size(): number {
+    return this._size;
+  }
+  private values: Object = {};
 
   delete(key: string): boolean {
-    this._values[key] = null;
+    this.values[key] = null;
+    this._size--;
     return true;
   }
 
   set(key: string, value: any): FastMap {
-    this._values[key] = value;
+    if (!this.get(key)) {
```

Comment:
I'm feeling this one's bit debatable since makes `set` slower.

- maybe let `size` simply returns `values.length` directly?
- or get rid of `size` interface?

It'll depends on usecase of `FastMap`, if we value efficiency of `get/set` operation, or expect `size` could be used frequently. So far, I haven't seen usage of `FastMap::size` in current codebase.